### PR TITLE
Fix Ruff rule B027 and B024

### PR DESCRIPTION
**Pull request by janasangeetha**
Fix Ruff rules
#6943

### CODE REVIEW

File: tfx/orchestration/portable/base_executor_operator.py [hunk 1]
```diff
@@ -84,6 +83,28 @@ def with_execution_watcher(
     self._execution_watcher_address = execution_watcher_address
     return self
 
+  @abc.abstractmethod
+  def handle_stop(self) -> None:
+    """Executor Operator specific logic to clean up after it is stopped."""
+    pass
+
+class BaseExecutorOperator(ParentBaseExecutorOperator):
```
Comment (lego0901) [inline tfx/orchestration/portable/base_executor_operator.py:91]: Is this really needed? I think it is sufficient to have `handle_stop` to be an abstractmethod, without needing to introduce `ParentBaseExecutorOperator`. I guess it is normal to have `handle_stop` implementation for any class that inherits `BaseExecutorOperator`.

File: tfx/tools/cli/handler/dag_runner_patcher.py [hunk 2]
```diff
@@ -56,11 +56,13 @@ def __init__(self, call_real_run=True):
     self._run_called = False
     self._call_real_run = call_real_run
 
+  @abc.abstractmethod
   def _before_run(self, runner: tfx_runner.TfxRunner,
                   pipeline: Union[pipeline_pb2.Pipeline, tfx_pipeline.Pipeline],
                   context: MutableMapping[str, Any]) -> None:
     pass
 
+  @abc.abstractmethod
```
Comment (lego0901) [inline tfx/tools/cli/handler/dag_runner_patcher.py:65]: Same here (and above `_before_run`). I believe all we need is `@abc.abstractmethod` decorators.

**Comment by lego0901** [comment]
SG for B024. Please tackle me if anything is wrong with my comments on B027. Thanks!

**Comment by janasangeetha** [comment]
@lego0901 On adding `@abc.abstractmethod` decorators, we observe failure in test cases #6942.
We should define the abstract methods of an abstract class in a child class. Hence, the parent classes are introduced.

**Comment by lego0901** [comment]
I see.. Thanks. Indeed those methods are intentionally no-ops unless they are overridden.
Is there any way to suppress those Ruff warnings similar to B024? They are clearly not abstract methods, but adding a unnecessary chain seems not good for maintenance.

File: tfx/types/system_artifacts.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ class SystemArtifact(abc.ABC):
   The subclasses, e.g, Dataset, Model, Statistics, e.t.c, match the MLMD types
   from third_party/ml_metadata/metadata_store/mlmd_types.py.
   """
-
+  # noqa: B024
```
Comment (lego0901) [inline tfx/types/system_artifacts.py:33]: This wasn't effective. Please try it out with

```python
class SystemArtifact(abc.ABC):  # noqa: B024
```

File: tfx/types/system_executions.py [hunk 4]
```diff
@@ -30,7 +30,7 @@ class SystemExecution(abc.ABC):
   The subclasses, e.g, Train, Transform, Process, e.t.c, match the MLMD types
   from third_party/ml_metadata/metadata_store/mlmd_types.py.
   """
-
+  # noqa: B024
```
Comment (lego0901) [inline tfx/types/system_executions.py:33]: ditto

File: tfx/orchestration/portable/base_executor_operator.py [hunk 5]
```diff
@@ -84,6 +83,28 @@ def with_execution_watcher(
...
+
+    Args:
+      execution_info: A wrapper of the info needed by this execution.
+
+    Returns:
+      The output from executor.
+    """
+    pass
+
   def handle_stop(self) -> None:
```
Comment (lego0901) [inline tfx/orchestration/portable/base_executor_operator.py:108]: You can suppress B027 check via

```python
  def handle_stop(self) -> None:  # noqa: B027
```

File: tfx/tools/cli/handler/dag_runner_patcher.py (hunk 2 above)
Comment (lego0901) [reply tfx/tools/cli/handler/dag_runner_patcher.py:65]: ditto

**Comment by lego0901** [comment]
Please make sure that everything is clear if you run
```
ruff check tfx --exclude tfx/examples/airflow_workshop/taxi/notebooks/*.ipynb --select "B024,B027"
```

Thanks!

File: tfx/orchestration/portable/base_executor_operator.py [hunk 6]
```diff
@@ -84,6 +84,6 @@ def with_execution_watcher(
     self._execution_watcher_address = execution_watcher_address
     return self
 
-  def handle_stop(self) -> None:
+  def handle_stop(self) -> None:# noqa: B027
```
Comment (lego0901) [inline tfx/orchestration/portable/base_executor_operator.py:87]: And also a nit comment; as a coding convention, we were using two spaces before any comment `#`. This was warned when we are on the internal Google land, but I am not sure if it is captured by the linter here.

@smokestacklightnin could you let me know if this comment style is acceptable by the linter? `pylint --rcfile ./pylintrc tfx/types/system_executions.py` didn't warn me

**Review by lego0901 — APPROVED** [review]
Thanks!

File: tfx/orchestration/portable/base_executor_operator.py (hunk 6 above)
Comment (smokestacklightnin) [reply tfx/orchestration/portable/base_executor_operator.py]: I don't believe the number of spaces matters. I tried with zero, one, and two spaces, and all three options passed `ruff check`.

**Pull request merged**